File: HoYa/Entities/Enquiry.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace HoYa.Entities
{
    public class Enquiry : Detail<EnquiryGeneral>//Process.Type.Value=="Enquiry"
    { 
        public virtual Guid? MaterialId { get; set; }
        [ForeignKey("MaterialId")]
        public virtual Material Material { get; set; }//原料耗材能源設備折舊成本
        public float? MinAmount { get; set; }//主管壓底價
        public float? Quantity { get; set; }
    }

    public class EnquiryGeneral : Extention
    {
        public virtual Guid? ProfileId { get; set; }
        [ForeignKey("ProfileId")]
        public virtual Profile Profile { get; set; }
        public virtual Guid? ProcessId { get; set; }
        [ForeignKey("ProcessId")]
        public virtual Process Process { get; set; }
        public string ContactPerson { get; set; }
        public string ContactValue { get; set; }
        public string CustomerName { get; set; }
        public string Content { get; set; }
    }
}

File: HoYa.Web/src/entities/enquiry.ts
import { Extention, Detail } from "./entity";
import { Process } from "./workflow";
import { Material } from "./material";
import { Profile } from "./person";
export class Enquiry extends Detail<EnquiryGeneral>
{
    materialId: string;
    material: Material;
    minAmount: number;
    quantity: number;
    price: number;
    priceMin: number;
    priceMax: number;
    constructor(ownerId?:string) {
        super();
        this.ownerId = ownerId;
        this.quantity = 1;
    }
}

export class EnquiryGeneral extends Extention {
    profileId: string;
    profile: Profile;
    processId: string;
    process: Process;
    contactPerson: string;
    contactValue: string;
    customerName: string;
    content: string;
    constructor() {
        super();
    }
}